django/template/defaultfilters.py

```diff
@@ -119,9 +119,21 @@ def floatformat(text, arg=-1):
     * {{ num2|floatformat:"-3" }} displays "34"
     * {{ num3|floatformat:"-3" }} displays "34.260"
 
+    Use the suffix ``g`` to force the result to be grouped
+    (appropriately for the locale, if settings.`USE_L10N` is True):
+
+    * {{ 6666.6666|floatformat:"2g" }} displays "6,666.67"
+    * {{ 10000|floatformat:"g" }} displays "10,000"
+
     If the input float is infinity or NaN, display the string representation
     of that value.
     """
+    force_grouping = False
+    if isinstance(arg, str):
+        force_grouping = arg.endswith('g')
+        arg = arg.strip('g')
```

Comment:
I think that we only want to remove a single `'g'` from the end of the string and continue to allow things like `'gg'`, `'1gg'`, `'g2'`, and `'g3g'` to raise an error as they would have before.
```suggestion
        if arg.endswith('g'):
            arg = arg[:-1]
            force_grouping = True
```